vsphere/stackstate_checks/vsphere/vsphere.py

```diff
@@ -1021,35 +1023,37 @@ def _vsphere_datacenters(self, content, domain="Unspecified", regexes=None, incl
             topology_tags = {}
             labels = []
             hostname = c.name
+            try:
+                if isinstance(c, vim.Datacenter):
+                    topology_tags["name"] = c.name
+                    topology_tags["id"] = c._moId
+                    topology_tags["layer"] = TOPOLOGY_LAYERS.DATACENTER
+                    topology_tags["domain"] = domain
+
+                    sts_identifiers, labels = self.extract_tags("Datacenter", c._moId)
+                    topology_tags["identifiers"] = sts_identifiers
 
-            if isinstance(c, vim.Datacenter):
-                datastores = []
-                for datastore in c.datastore:
-                    datastores.append(datastore.name)
-                topology_tags["topo_type"] = VSPHERE_COMPONENT_TYPE.DATACENTER
-                topology_tags["datastores"] = datastores
-                topology_tags["name"] = c.name
-                topology_tags["id"] = c._moId
-                topology_tags["layer"] = TOPOLOGY_LAYERS.DATACENTER
-                topology_tags["domain"] = domain
-
-                sts_identifiers, labels = self.extract_tags("Datacenter", c._moId)
-                topology_tags["identifiers"] = sts_identifiers
-
-                computeresources = []
-                clustercomputeresources = []
-
-                for computeres in c.hostFolder.childEntity:
-                    if isinstance(computeres, vim.ComputeResource):
-                        computeresources.append(computeres.name)
-                    elif isinstance(computeres, vim.CloudComputeResource):
-                        clustercomputeresources.append(computeres.name)
-
-                topology_tags["computeresources"] = computeresources
-                topology_tags["clustercomputeresources"] = clustercomputeresources
-
-                add_label_pair(labels, "name", topology_tags["name"])
-                hostname = None
+                    computeresources = []
+                    clustercomputeresources = []
+                    datastores = []
+                    for datastore in c.datastore:
+                        datastores.append(datastore.name)
+                    topology_tags["topo_type"] = VSPHERE_COMPONENT_TYPE.DATACENTER
```

Comment:
I would set the type above with the other required parameters (name, id, layer, etc.). Otherwise, if an error occurs in the code between setting the identifiers and the type, we could end up with a component that misses a type.